server/app/src/test/java/com/google/heartrate/wearos/app/gatt/heartrate/characteristics/HeartRateGattCharacteristicManagerTest.java

```diff
@@ -0,0 +1,159 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = {Build.VERSION_CODES.O_MR1}, manifest = Config.NONE)
+public class HeartRateGattCharacteristicManagerTest {
+
+    private HeartRateMeasurementCharacteristic characteristic;
+
+    @Before
+    public void setUp() {
+        characteristic = new HeartRateMeasurementCharacteristic();
+    }
+
+    /**
+     * Assent setting values to given Heart Rate Characteristic was successful.
+     * <br/>Assents no exceptions was thrown and after set Heart Rate Value and Expended Energy Value
+     * are equal to requested.
+     *
+     * @param characteristic      characteristic to set value
+     * @param heartRateValue      heart rate value to set
+     * @param expendedEnergyValue expended energy value to set
+     */
+    public static void assertSetValuesSuccessful(HeartRateMeasurementCharacteristic characteristic,
+                                                 int heartRateValue, int expendedEnergyValue) {
+        final int[] currentHeartRateMeasurementValue = new int[1];
+        final int[] currentExpendedEnergyValue = new int[1];
+
+        assertNotThrows(() -> characteristic
+                .setHeartRateCharacteristicValue(heartRateValue, expendedEnergyValue));
+        assertNotThrows(() -> currentHeartRateMeasurementValue[0] = characteristic
+                .getHeartRateMeasurementValue());
+        assertNotThrows(() -> currentExpendedEnergyValue[0] = characteristic
+                .getExpandedEnergyValue());
+
+        assertEquals(heartRateValue, currentHeartRateMeasurementValue[0]);
+        assertEquals(expendedEnergyValue, currentExpendedEnergyValue[0]);
+    }
+
+    /**
+     * Assent set values to given Heart Rate Characteristic fails with GattException.
+     *
+     * @param characteristic      characteristic to set value
+     * @param heartRateValue      heart rate value to set
+     * @param expendedEnergyValue expended energy value to set
+     */
+    public static void assertSetValuesFailed(HeartRateMeasurementCharacteristic characteristic,
+                                             int heartRateValue, int expendedEnergyValue) {
+        assertThrows(GattException.class, () -> characteristic
+                .setHeartRateCharacteristicValue(heartRateValue, expendedEnergyValue));
+    }
+
+    /**
+     * Assent failed set values to given Heart Rate Characteristic not change stored values.
+     *
+     * @param characteristic      characteristic to set value
+     * @param heartRateValue      heart rate value to set
+     * @param expendedEnergyValue expended energy value to set
+     */
+    public static void assertValuesNotChangedAfterFailedSet(HeartRateMeasurementCharacteristic characteristic,
+                                                            int heartRateValue, int expendedEnergyValue) {
+        final int[] beforeSetHeartRateMeasurementValue = new int[1];
```

Comment:
Why arrays?

Reply:
Fixed in [commit](https://github.com/googleinterns/heartrate-bt-wear/pull/1/commits/93a29ed3ea6d7a2bf9235ec829a41d0e44889a43)